### ISSUE TITLE
Regarding modifying code to perform Mann Kendall and Sen's trend analysis

### ISSUE THREAD

**Issue by cicatrixx** [ARCCSS-extremes/climpact2#62]
For our analysis, we want to perform the Mann Kendall and Sens trend analysis tests. So I am looking inside the codes in `climpact2.GUI.r` to figure out if it would be easy to change the linear fit test codes to do this. My understanding is that the linear trend evaluation is done inside function `plotx` in the following lines for code where the linear trend fit output is saved as a global variable `out`. 
```
  fit <- suppressWarnings(lsfit(1:length(x), y))		# assumes time intervals are always evenly spaced
  out <<- ls.print(fit, print.it = FALSE)
  r2 <- round(100 * as.numeric(out$summary[1, 2]), 1)
  pval <- round(as.numeric(out$summary[1, 6]), 3)
  beta <- round(as.numeric(out$coef.table[[1]][2, 1]), 3)
  betaerr <- round(as.numeric(out$coef.table[[1]][2, 2]), 3)
```
The `out` variable is then called inside the `index.calc` function to save the results to a csv file as:
```
cat(file=trend_file,paste(tmp.index.name,metadata$year.start,metadata$year.end,round(as.numeric(out$coef.table[[1]][2, 1]), 3),round(as.numeric(out$coef.table[[1]][2, 2]), 3),round(as.numeric(out$summary[1, 6]),3),sep=","),fill=180,append=T)
```

But there are also functions `plot.hw` and `plot.precindex` that are used for creating plots special indices. These functions both call `plotx` and calculate linear trend fit again with output saved as a global variable `out1`. However, when saving trend output to csv file, both these function use variable `out`. 
```
plotx((date.years), index[def,asp,], main = gsub('\\*', unit, plot.title),ylab = unit,xlab = x.label,index.name=index.name,sub=sub)
#			dev.copy()
dev.off(dev0)      
fit1<-suppressWarnings(lsfit((date.years),index[def,asp,]))
out1<<-ls.print(fit1,print.it=F)  cat(file=trend_file,paste(paste(definitions[def],aspects[asp],sep="."),metadata$year.start,metadata$year.end,round(as.numeric(out$coef.table[[1]][2, 1]), 3),round(as.numeric(out$coef.table[[1]][2, 2]), 3),round(as.numeric(out$summary[1, 6]),3),sep=","),fill=180,append=T)

```
It seems that `out1` is redundant here but is there something I am missing? Should the `cat` statement inside `plot.hw` and `plot.precindex` have called `out1` instead of `out` because the special indices have to be handled differently? Is this just a typo in the code or can `out1`  be ignored entirely?

**Comment by heroldn** [comment]
It looks like the 'out1' lines you've pointed out are redundant, as the plot.hw and plot.precindex functions call plotx anyway, which as you say calculates 'out'. I will remove these lines, thanks for pointing this out. There is still cleaning up to do in this code.

I had a look at implementing Mann Kendall some time ago but I believe it required an additional package and the preference was to not add to any more of the dependencies climpact has. This decision might/can change so happy to hear peoples' preferences.

**Comment by cicatrixx** [comment]
Thanks for the clarification! I vote for having a more robust trend test than linear fit would be better. In our literature review, we found that most recent literature use at least Mann Kendall and Sen Slope tests. Also note that RClimDex developers also removed linear fitting feature from the package in the last years to encourage people to do more robust trend tests. 

I was able to modify the code to additionally conduct the Mann Kendall and Sen Slope tests using the trend package. I can share the code once I have a cleaned it up in case anyone is interested in doing so before you implement it officially.

**Comment by cicatrixx** [comment]
@heroldn Do you have any advice on how missing values should be handled for Mann Kendall and Sen Slope tests? I had simply skipped them in my tests but now I am not sure if that is reasonable if there are consecutive years or months of missing data.

**Comment by cicatrixx** [comment]
@heroldn Do you have any advice on how missing values should be handled for Mann Kendall and Sen Slope tests? I had simply skipped them in my tests but now I am not sure if that is reasonable if there are consecutive years or months of missing data.

**Comment by cicatrixx** [comment]
In attached file, I have implemented the Mann Kendall Test (MKT) and Sens Slope (SS) calculation using the trend package. The package cannot handle missing data, so time steps with missing values are just removed before calculation. I checked the Kendall package and found that they do the same to handle missing data. I also cross-checked the MKT calculations from trend and Kendall packages. Changes I have made in the file are noted at the top.

[climpact2.GUI_wMLTandSStest.zip](https://github.com/ARCCSS-extremes/climpact2/files/2790263/climpact2.GUI_wMLTandSStest.zip)
